ibis/pyspark/compiler.py

```diff
@@ -1052,32 +1055,53 @@ def compile_window_op(t, expr, scope, timecontext, **kwargs):
     ]
 
     order_by = window._order_by
+    # Timestamp needs to be cast to long for filtering in spark
     ordering_keys = [
-        key.to_expr().get_name()
+        F.col(key.to_expr().get_name()).cast('long')
```

Comment:
This looks really weird. Why are we getting op from expr and then use `to_expr` to get expr from op again?

Reply:
```
ordering_keys = [
   F.col(expr.get_name()).cast('long')
   if isinstance(expr, types.TimestampColumn)
   else expr.get_name()
   for expr in order_by
]
```
Does this work?

Reply:
This doesn't work but I get your point that we don't need to call `op` and `to_expr` at the same time. I refactored a little.